Content.Server/Implants/ChameleonControllerSystem.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Content.Server.Clothing.Systems;
+using Content.Server.Preferences.Managers;
+using Content.Shared.Clothing;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory;
+using Content.Shared.Preferences;
+using Content.Shared.Preferences.Loadouts;
+using Content.Shared.Roles;
+using Content.Shared.Station;
+using Content.Shared.Timing;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Implants;
+
+public sealed class ChameleonControllerSystem : SharedChameleonControllerSystem
+{
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly SharedStationSpawningSystem _stationSpawningSystem = default!;
+    [Dependency] private readonly ChameleonClothingSystem _chameleonClothingSystem = default!;
+    [Dependency] private readonly IServerPreferencesManager _preferences = default!;
+    [Dependency] private readonly UseDelaySystem _delay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SubdermalImplantComponent, ChameleonControllerSelectedOutfitMessage>(OnSelected);
+
+        SubscribeLocalEvent<ChameleonClothingComponent, InventoryRelayedEvent<ChameleonControllerOutfitSelectedEvent>>(ChameleonControllerOutfitItemSelected);
+    }
+
+    private void OnSelected(Entity<SubdermalImplantComponent> ent, ref ChameleonControllerSelectedOutfitMessage args)
+    {
+        if (!_delay.TryResetDelay(ent.Owner, true) || ent.Comp.ImplantedEntity == null || !HasComp<ChameleonControllerImplantComponent>(ent))
+            return;
+
+        ChangeChameleonClothingToJob(ent.Comp.ImplantedEntity.Value, args.SelectedChameleonOutfit);
+    }
+
+    /// <summary>
+    ///     Switches all the chameleon clothing that the implant user is wearing to look like the selected job.
+    /// </summary>
+    private void ChangeChameleonClothingToJob(EntityUid user, ProtoId<ChameleonOutfitPrototype> outfit)
+    {
+        var outfitPrototype = _proto.Index(outfit);
+
+        _proto.TryIndex(outfitPrototype.Job, out var jobPrototype);
+        _proto.TryIndex(outfitPrototype.StartingGear, out var startingGearPrototype);
+
+        RoleLoadout? customRoleLoadout = null;
+        RoleLoadout? defaultRoleLoadout = null;
+        StartingGearPrototype? jobStartingGearPrototype = null;
+
+        if (jobPrototype != null)
+        {
+            _proto.TryIndex(jobPrototype.StartingGear, out jobStartingGearPrototype);
+
+            if (!TryComp<ActorComponent>(user, out var actorComponent))
+                goto slotIterator;
```

Comment:
I can't say for certain if `goto` is OK practice or not, but I am like 70% sure it's spooky (using `if` statements or separate it out into functions would be the way to go I think?)

Seek feedback from a more experienced programmer here.